src/components/editor/editor.js

```diff
@@ -333,7 +350,28 @@ class PageContainer extends React.Component {
       );
     };
 
-    
+  surroundText = (ends) => {
+    const editorState = this.state.editorState;
+    // get selection
+    let selectionState = editorState.getSelection();
+    const anchorKey = selectionState.getAnchorKey();
+    let currentContent = editorState.getCurrentContent();
+    const currentContentBlock = currentContent.getBlockForKey(anchorKey);
+    const start = selectionState.getStartOffset();
+    const end = selectionState.getEndOffset();
+    const selectedText = currentContentBlock.getText().slice(start, end);
+    currentContent = Modifier.replaceText(
+      currentContent,
+      selectionState,
+      `${ends}${selectedText}${ends}`
+    );
+
+    this.setState({
+      editorState: EditorState.push(editorState, currentContent, 'insert-characters')
+    });
+  };
```

Comment:
I like this concept, but one thing that confuses me is when I hit ctrl + b in the middle (or at the start of a line) expecting that line to start bold I get four stars behind the cursor. I think in that case, it would make sense to start the cursor in the middle of the stars. ctrl+b with no selection made -> **|**

Reply:
oh, I see now that I played myself above. there should be two stars on either side of that | but markdown wouldn't allow it.

Reply:
I was trying for a very long time to get the cursor to land in the middle but couldn't get the selectionState to do what I wanted. If you know how to do it, please let me know what the process is, because I know I'll need to use it again in the future.

Reply:
Check out https://draftjs.org/docs/api-reference-editor-state/#forceselection 

You could have it insert the characters, use a getSelectionAfter to get the selection after the characters are added (in the callback to the insert-characters set state), then modify the selection object to put the cursor (I think using the offset property?) where you want it, then EditorState.forceSelection() with the new selection state. Then you need to update the state with the new editor state that's created by using forceSelection(). Not 100% sure that's all correct but something pretty close to that should work.

Reply:
No prob. Looks like a good solution to me. This feature tests good. 

Go ahead and merge in the changes from master (I've since merged in a feature) and give it a good testing just to be safe.

If it all works, go ahead and merge.